VogtPayroll2/PayrollConsoleReader.cs

```diff
@@ -0,0 +1,139 @@
+            int hoursWorked;
+            decimal salary;
+
+            name = GetNameConsole();
+            hoursWorked = GetHoursWorkedConsole();
+            salary = GetSalaryConsole();
+
+            return new SalaryEmployee(name, hoursWorked, salary);
+
+        }
+        #endregion
+
+        #region ReadHourlyEmployeeFromConsole
+        /// <summary>
+        /// Create an hourly employee
+        /// </summary>
+        /// <returns>An hourly employee</returns>
+        public HourlyEmployee ReadHourlyEmployeeFromConsole()
+        {
+            string name;
+            int hoursWorked;
+            decimal payrate;
+
+            name = GetNameConsole();
+            hoursWorked = GetHoursWorkedConsole();
+            payrate = GetPayrateConsole();
+
+            return new HourlyEmployee(name, payrate, hoursWorked);
+
+        }
+        #endregion
+
+        #region GetPayrateConsole
+        /// <summary>
+        /// Gets the payrate from user
+        /// </summary>
+        /// <returns>The payrate</returns>
+        public decimal GetPayrateConsole()
+        {
+            decimal payrateResult;
+            Console.WriteLine("What is the payrate amount?");
+            var payrate = Console.ReadLine();
+            while (!decimal.TryParse(payrate, out payrateResult))
+            {
+                Console.WriteLine("Not a payrate amount!");
+                payrate = Console.ReadLine();
+            }
+
+            return payrateResult;
+
+        }
+        #endregion
+
+        #region GetHoursWorkedConsole
+        /// <summary>
+        /// Gets the hours worked from user
+        /// </summary>
+        /// <returns>The number of hours worked</returns>
+        public int GetHoursWorkedConsole()
+        {
+
+            int hoursResult;
+            Console.WriteLine("How many hours were worked?");
+            var hoursWorked = Console.ReadLine();
+            while (!int.TryParse(hoursWorked, out hoursResult))
+            {
+                Console.WriteLine("Not a number!");
+                hoursWorked = Console.ReadLine();
+            }
+
+            return hoursResult;
+
+        }
+        #endregion
+
+        #region GetNameConsole
+        /// <summary>
+        /// Gets the name from user
+        /// </summary>
+        /// <returns>The name</returns>
+        public string GetNameConsole()
+        {
+            Console.WriteLine("What is the name of the employee?");
+
+            var name = Console.ReadLine();
+
+            while (string.IsNullOrEmpty(name))
+            {
+                Console.WriteLine("The name can't be empty! try again");
+                name = Console.ReadLine();
+            }
+
+            return name;
+
+        }
+        #endregion
+
+        #region GetSalaryConsole
+        /// <summary>
+        /// Gets the salary from user
+        /// </summary>
+        /// <returns>The salary</returns>
+        public decimal GetSalaryConsole()
```

Comment:
This works, but by reusable functions I meant more something like:

```c#
public decimal GetDecimalFromConsole() {
              decimal value;

            var input = Console.ReadLine();

            while (!decimal.TryParse(input, out result))
            {
                Console.WriteLine("Not a decimal!");
                value = Console.ReadLine();
            }

            return value;
}
```

So you could then reuse it anywhere like this:
```c#
public SalaryEmployee ReadSalaryEmployeeFromConsole() {
            Console.WriteLine("What is the salary of the employee?");
            decimal salary = GetDecimalFromConsole();
}
```